core/app/components/pubs/InitialCreatePubForm.tsx

```diff
@@ -0,0 +1,272 @@
+		</>
+	);
+};
+
+const RelatedPubValueSelector = ({
+	sourcePubId,
+	pubField,
+}: {
+	// Only used for FileUpload, which we want to upload to the source pub
+	sourcePubId: PubsId;
+	pubField: Props["relatedPubFields"][number] | undefined;
+}) => {
+	if (!pubField) {
+		return null;
+	}
+	const element = makeFormElementDefFromPubFields([pubField])[0] as PubFieldElement;
+	return (
+		<ConfigureRelatedValue
+			element={element}
+			pubId={sourcePubId}
+			// @ts-ignore TODO: how best to fix this?
+			slug="relatedPub.value"
+			values={[]}
+			className="w-fit"
+		/>
+	);
+};
+
+const baseSchema = Type.Object({
+	pubTypeId: Type.String(),
+});
+const schemaWithRelatedPub = Type.Object({
+	pubTypeId: Type.String(),
+	relatedPub: Type.Object({
+		relatedPubId: Type.String(),
+		slug: Type.String(),
+		value: Type.Any(),
+	}),
+});
+
+interface Props {
+	pubTypes: Pick<PubTypes, "id" | "name">[];
+	relatedPubFields: Pick<PubField, "id" | "slug" | "name" | "schemaName">[];
+	stageId?: StagesId;
+	relatedPubId?: PubsId;
+}
+/** The first step in creating a pub—choosing a pub type, and possibly a related pub */
+export const InitialCreatePubForm = ({
+	pubTypes,
+	relatedPubFields,
+	stageId,
+	relatedPubId,
+}: Props) => {
+	const { defaultValues } = useMemo(() => {
+		const defaultValues = { pubTypeId: undefined };
+		if (relatedPubId) {
+			return {
+				defaultValues: {
+					...defaultValues,
+					relatedPub: {
+						relatedPubId,
+					},
+				},
+			};
+		}
+		return { schema: baseSchema, defaultValues };
+	}, [relatedPubId]);
+
+	const form = useForm<typeof baseSchema | typeof schemaWithRelatedPub>({
+		mode: "onChange",
+		reValidateMode: "onChange",
+		defaultValues,
+		resolver: (values, context, options) => {
+			let schema = baseSchema;
+			if (relatedPubId) {
+				const slug = values.relatedPub.slug;
+				const pubField = relatedPubFields.find((pf) => pf.slug === slug);
+				// The same as `schemaWithRelatedPub` but with the `value` schema dynamically generated based on the chosen field
+				schema = Type.Object({
+					pubTypeId: Type.String(),
+					relatedPub: Type.Object({
+						relatedPubId: Type.String(),
+						slug: Type.String(),
+						value: pubField?.schemaName
+							? getJsonSchemaByCoreSchemaType(pubField.schemaName)
+							: Type.Any(),
+					}),
+				});
+			}
+			const createResolver = typeboxResolver(schema);
+			return createResolver(values, context, options);
+		},
+	});
+
+	const path = usePathname();
+	const searchParams = useSearchParams();
+	const router = useRouter();
+	const community = useCommunity();
+
+	const pathWithoutFormParam = useMemo(() => {
+		const urlSearchParams = new URLSearchParams(searchParams ?? undefined);
+		urlSearchParams.delete("create-pub-form");
+		return `${path}?${urlSearchParams.toString()}`;
+	}, [path, searchParams]);
+
+	const closeForm = useCallback(() => {
+		router.replace(pathWithoutFormParam);
+	}, [pathWithoutFormParam]);
```

Comment:
all of this is from the original `PubTypeFormClient` which I renamed since this does more than just pub type now